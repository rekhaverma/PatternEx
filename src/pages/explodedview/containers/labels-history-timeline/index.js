export { default } from './labels-history-timeline.container.jsx';
