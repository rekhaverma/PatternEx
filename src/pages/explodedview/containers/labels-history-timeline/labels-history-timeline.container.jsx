import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { compose } from 'redux';
import { FormattedMessage, FormattedHTMLMessage } from 'react-intl';

import { Button } from 'components/forms';
import NoData from 'components/no-data';
import Modal from 'components/modal';
import ChangeLabel from 'containers/change-label';
import { findEntityName, getEntityTypeByPipeline } from 'lib/decorators';
import { getSourceLog } from 'lib';

import Section from '../../components/section';
import TimelineChart from '../../components/timelinechart';

import {
  withSearchData,
  withLabelsHistoryData,
  withLabelsActions,
  withLocationData,
  withHistoricalData,
} from '../../hoc';
import { entityNodebookID } from '../../config';

export class LabelsHistoryTimeline extends Component {
  constructor(props) {
    super(props);

    this.state = {
      'predictionRow': {},
      'changeLabelModal': false,
      'openModal': false,
    };

    this.handleInvestigate = this.handleInvestigate.bind(this);
    this.handleSetLabel = this.handleSetLabel.bind(this);
    this.setLabelActionHandler = this.setLabelActionHandler.bind(this);
    this.handleVirusScan = this.handleVirusScan.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    const { labelsHistory, searchData } = nextProps;
    const entityDate = moment(searchData.ts);
    let labelFound = false;
    for (let i = 0; i < labelsHistory.length; i += 1) {
      const labelDate = moment(labelsHistory[i].create_time);
      if (labelDate.isSame(entityDate, 'day')) {
        labelFound = true;
      }
    }
    this.props.changeSetLabelDisplay(labelFound);
  }

  setLabelActionHandler(row) {
    const { location } = this.props;
    const behaviorType = location.behavior_type;
    const getLabels = () => {
      this.props.getEntityLabelHistory(location);
      this.props.getSearchData(location);
      this.props.getHistoricalData(location);
      this.props.changeSetLabelDisplay(true);
    };

    const params = row;
    if (behaviorType !== 'suspicious' && behaviorType !== 'malicious') {
      if (!params.mode) {
        params.mode = location.mode;
      }
      if (!params.model_name) {
        params.model_name = location.model_name;
      }
    }
    this.props.setLabelHandler(row, behaviorType, getLabels);
  }


  handleVirusScan() {
    const { location } = this.props;
    const baseUrl = 'https://www.virustotal.com/#/';
    let url = `${baseUrl}home/search/`;

    if (location.pipeline === 'domain' || location.pipeline === 'sipdomain') {
      let domain;

      switch (location.pipeline) {
        case 'domain':
          domain = location.entity_name;
          break;
        case 'sipdomain':
          const dataSplit = location.entity_name.split(' ');
          domain = dataSplit[1];
          break;
        default:
          domain = '';
      }

      url = `${baseUrl}domain/${domain}`;
    }

    const win = window.open(url, '_blank');
    win.focus();
  }

  handleInvestigate() {
    const { searchData, location } = this.props;
    const pipeline = location.pipeline;
    const sourceLog = getSourceLog(pipeline, searchData);

    const entityName = `entity_name=${findEntityName(pipeline, searchData)}`;
    const entityType = `entity_type=${getEntityTypeByPipeline(pipeline)}`;
    const customerName = `customer_name=${this.props.customerName}`;
    const date = `date=${moment(this.props.searchData.ts).format('MM/DD/YYYY hh:mm A')}`;
    const sourceStr = `source=${sourceLog}`;
    const pipelineStr = `pipeline=${pipeline}`;

    const investigateURL = `/analytics/#/notebook/${entityNodebookID}?${customerName}&${entityName}&${entityType}&${pipelineStr}&${date}&${sourceStr}`;
    const win = window.open(investigateURL, '_blank');
    win.focus();
  }

  handleSetLabel(modal = true) {
    if (!this.props.setLabelDisabled) {
      const { searchData, location } = this.props;
      const { pipeline } = location;
      const newState = {
        'changeLabelModal': modal,
        'openModal': modal,
        'predictionRow': this.state.predictionRow,
      };

      if (modal) {
        newState.predictionRow = {
          ...searchData,
          pipeline,
          method_name: location.method_name,
          id: location.entity_id,
          tag_id: searchData.predicted_tag_id,
          entity_type: getEntityTypeByPipeline(pipeline),
        };
      }

      this.setState(newState);
    }
  }

  render() {
    const { startDate, endDate } = this.props;

    return (
      <Section
        size="medium"
        className="+no-margin +overflow-visible"
        loaded
        isDraggable={this.props.isDraggable}
      >
        <div className="labelTimelineTop">
          <div className="dateRange">
            <div className="dateRange__presentation">
              <div className="dateRange__presentation__column">
                <span>From: <span className="dateRangeDate">{moment(startDate).format('MM - DD - YYYY')}</span></span>
                <span>Until: <span className="dateRangeDate">{moment(endDate).format('MM - DD - YYYY')}</span></span>
              </div>
            </div>
          </div>
          <span className="icon-drag" />
          <div>
            <Button
              className="button--success +small"
              disabled={this.props.setLabelDisabled}
              onClick={() => this.handleSetLabel(true)}
            >
                Set Label
            </Button>
            <Button
              className="button--dark +small"
              onClick={this.handleVirusScan}
            >
                Virus Total
            </Button>
            <Button
              className="button--dark +small"
              onClick={this.handleInvestigate}
            >
                Investigate
            </Button>
          </div>
        </div>

        <div>
          { this.props.labelsHistory.length <= 0 &&
          <NoData
            intlId="global.nodata"
            intlDefault="There is no data to display"
            className="nodata--timeline"
            withIcon
          />
          }
          <TimelineChart
            data={this.props.labelsHistory}
            startDate={startDate}
            endDate={endDate}
          />
        </div>

        {
          this.state.openModal && (
            <Modal>
              { this.state.changeLabelModal &&
              <ChangeLabel
                predictionRow={this.state.predictionRow}
                onCancel={() => this.handleSetLabel(false)}
                onSave={this.setLabelActionHandler}
                title={findEntityName(this.state.predictionRow.pipeline, this.state.predictionRow) !== ''
                  ? <FormattedHTMLMessage
                    id="evp.setLabel"
                    values={{ 'entityName': findEntityName(this.state.predictionRow.pipeline, this.state.predictionRow) }}
                  />
                  : <FormattedMessage id="evp.setLabel" />}
              />
              }
            </Modal>
          )
        }
      </Section>
    );
  }
}

LabelsHistoryTimeline.displayName = 'LabelsHistoryTimeline';

LabelsHistoryTimeline.propTypes = {
  // 'isLabelsHistoryLoaded': PropTypes.bool,
  'startDate': PropTypes.object,
  'endDate': PropTypes.object,
  'labelsHistory': PropTypes.array,
  'searchData': PropTypes.object,
  'location': PropTypes.object,
  'customerName': PropTypes.string,
  'setLabelHandler': PropTypes.func,
  'getSearchData': PropTypes.func,
  'getHistoricalData': PropTypes.func,
  'getEntityLabelHistory': PropTypes.func,
  'setLabelDisabled': PropTypes.bool,
  'changeSetLabelDisplay': PropTypes.func,
  'isDraggable': PropTypes.bool,
};

LabelsHistoryTimeline.defaultProps = {
  // 'isLabelsHistoryLoaded': false,
  'startDate': moment(),
  'endDate': moment(),
  'labelsHistory': [],
  'searchData': {},
  'location': {},
  'customerName': '',
  'setLabelHandler': () => {},
  'getSearchData': () => {},
  'getHistoricalData': () => {},
  'getEntityLabelHistory': () => {},
  'setLabelDisabled': false,
  'changeSetLabelDisplay': () => {},
  'isDraggable': true,
};

const enhance = compose(
  withHistoricalData,
  withSearchData,
  withLabelsHistoryData,
  withLocationData,
  withLabelsActions,
);
export default enhance(LabelsHistoryTimeline);
