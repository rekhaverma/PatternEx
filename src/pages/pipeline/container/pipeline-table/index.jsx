export { default } from './pipeline-table.container';
