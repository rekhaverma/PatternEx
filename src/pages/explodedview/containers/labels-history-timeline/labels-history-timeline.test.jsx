import React from 'react';
import moment from 'moment';

import { LabelsHistoryTimeline } from './labels-history-timeline.container';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<LabelsHistoryTimeline />', () => {
  describe('render: ', () => {
    it('should match with snapshot', () => {
      const props = {
        startDate: moment.utc('2018-05-24T00:00:00.000Z'),
        endDate: moment.utc('2018-06-23T23:59:59.999Z'),
        searchData: {},
        isLabelsHistoryLoaded: true,
        labelsHistory: [
          {
            name: 'CC Fraud',
            create_time: 'Sat, 23 Jun 2018 09:03:00 -0000',
            severity: 4,
            description: 'Credit Card Fraud. Account created using compromised/stolen credit card details and personal identity information',
            type: 'M',
            id: '98ca552c-bff2-4969-9d18-bcc4da4759f1',
          },
          {
            name: 'CC Fraud',
            create_time: 'Sat, 23 Jun 2018 09:03:00 -0000',
            severity: 4,
            description: 'Credit Card Fraud. Account created using compromised/stolen credit card details and personal identity information',
            type: 'M',
            id: 'd7f73c05-3187-41d1-93c8-2a72a44d892d',
          },
          {
            name: 'Exfiltration',
            create_time: 'Mon, 04 Jun 2018 08:57:00 -0000',
            severity: 4,
            description: 'Techniques and attributes that result or aid in the adversary removing files and information from a target network.',
            type: 'M',
            id: '679908c1-3b89-42b4-97bd-57d7e8655649',
          }],
        location: {
          behavior_type: 'suspicious',
          end_time: '06-23-2018',
          entity_id: '3a627a00-76c4-11e8-8caf-6bff10ab57c7',
          entity_name: '172.18.16.236 8.8.8.8',
          method_name: 'ranking',
          mode: 'realtime',
          model_name: '2018-04-29-Ranking-SipDip-1-ICMPEXFIL',
          pipeline: 'sipdip',
          start_time: '06-23-2018',
        },
        customerName: '',
      };
      const wrapper = shallow(<LabelsHistoryTimeline {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
