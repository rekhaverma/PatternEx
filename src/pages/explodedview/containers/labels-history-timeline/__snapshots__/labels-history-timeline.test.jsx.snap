// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelsHistoryTimeline /> render:  should match with snapshot 1`] = `
<Section
  className="+no-margin +overflow-visible"
  dropDownOptions={Array []}
  dropDownState={false}
  dropdown={false}
  expandable={false}
  expandableView={null}
  filterFeatureList={[Function]}
  isDraggable={true}
  loaded={true}
  modalStyle={Object {}}
  onDropDownClick={[Function]}
  onDropDownSelection={[Function]}
  onModalOpen={[Function]}
  selectedDropDown={Array []}
  size="medium"
  title={null}
>
  <div
    className="labelTimelineTop"
  >
    <div
      className="dateRange"
    >
      <div
        className="dateRange__presentation"
      >
        <div
          className="dateRange__presentation__column"
        >
          <span>
            From: 
            <span
              className="dateRangeDate"
            >
              05 - 24 - 2018
            </span>
          </span>
          <span>
            Until: 
            <span
              className="dateRangeDate"
            >
              06 - 23 - 2018
            </span>
          </span>
        </div>
      </div>
    </div>
    <span
      className="icon-drag"
    />
    <div>
      <Button
        className="button--success +small"
        disabled={false}
        onClick={[Function]}
      >
        Set Label
      </Button>
      <Button
        className="button--dark +small"
        onClick={[Function]}
      >
        Virus Total
      </Button>
      <Button
        className="button--dark +small"
        onClick={[Function]}
      >
        Investigate
      </Button>
    </div>
  </div>
  <div>
    <TimelineChart
      data={
        Array [
          Object {
            "create_time": "Sat, 23 Jun 2018 09:03:00 -0000",
            "description": "Credit Card Fraud. Account created using compromised/stolen credit card details and personal identity information",
            "id": "98ca552c-bff2-4969-9d18-bcc4da4759f1",
            "name": "CC Fraud",
            "severity": 4,
            "type": "M",
          },
          Object {
            "create_time": "Sat, 23 Jun 2018 09:03:00 -0000",
            "description": "Credit Card Fraud. Account created using compromised/stolen credit card details and personal identity information",
            "id": "d7f73c05-3187-41d1-93c8-2a72a44d892d",
            "name": "CC Fraud",
            "severity": 4,
            "type": "M",
          },
          Object {
            "create_time": "Mon, 04 Jun 2018 08:57:00 -0000",
            "description": "Techniques and attributes that result or aid in the adversary removing files and information from a target network.",
            "id": "679908c1-3b89-42b4-97bd-57d7e8655649",
            "name": "Exfiltration",
            "severity": 4,
            "type": "M",
          },
        ]
      }
      endDate={"2018-06-23T23:59:59.999Z"}
      startDate={"2018-05-24T00:00:00.000Z"}
    />
  </div>
</Section>
`;
